### PR TITLE
update ros launch files and nodes for update of rotors_simulator

### PR DESCRIPTION
**Pull request by thomasgubler**
this also removes the duplicated joy node.

rotors_simulator PR: https://github.com/PX4/euroc_simulator/pull/9
mav_comm PR: https://github.com/PX4/mav_comm/pull/1

supersedes #1852  (is the same but against master)

### CODE REVIEW

**Comment by thomasgubler** [comment]
@AndreasAntener a test would be very much appreciated

**Comment by AndreasAntener** [comment]
I can wait #1847 on this if it makes sense to respect the new NS arg for the mavros include too. I'm not so sure anymore if mavros has to be in root necessarily. Could there be multiple mavros nodes?

**Comment by thomasgubler** [comment]
Actually I did not update mavros_sitl.launch because of #1847. Well yes there could be multiple mavros nodes. One for each onboard computers (e.g. if you simulate multiple on one system). If me move it to ardrone/mavros (instead of /mavros or /px4_multicopter/mavros) this scenario becomes feasible.

**Comment by LorenzMeier** [comment]
@AndreasAntener @thomasgubler Can this go in now?

**Comment by thomasgubler** [comment]
@AndreasAntener could you have a quick look here? I updated the namespaces and some parameters. If I run the tests locally on my machine I get a 'Success'

**Comment by AndreasAntener** [comment]
I should have time tonight to have a look. I already tried to change all the launch files and scripts this week, but I'm running in the same issue as SITL CI currently has (that's why I haven't sent you a PR for it..).

Should I also do a test with
rotors_simulator PR: PX4/euroc_simulator#9
mav_comm PR: PX4/mav_comm#1
?

**Comment by thomasgubler** [comment]
> Should I also do a test with
rotors_simulator PR: PX4/euroc_simulator#9
mav_comm PR: PX4/mav_comm#1

ah of course that's the problem: CI is not using these branches which are required. As soon as the simulator is open source we need submodules to get the dependencies under control. Or we could even do one now: a new repo px4ros with the submodules: Firmware, rotors-simulator, mav-comm, ...

**Comment by AndreasAntener** [comment]
A new repo might make sense. For CI anyway. I believe catkin cannot resolve nested packages. So submodules might make things difficult there.

**Comment by thomasgubler** [comment]
the rotors_simulator is nested as well. The question is if more than one
level is supported.

**Comment by AndreasAntener** [comment]
euroc has multiple adjacent packages, but not nested ones. Nested would mean that the top-level folder has CMakeLists.txt and package.xml too

**Comment by thomasgubler** [comment]
@AndreasAntener ok yes nested as in packages does not work:
http://wiki.ros.org/ROS/Tutorials/catkin/CreatingPackage#What_makes_up_a_catkin_Package.3F

@LorenzMeier, we might need to reconsider the :umbrella: repository solution. Still I don't know if multiple levels of directory nesting will work (I guess so...)

**Comment by thomasgubler** [comment]
I just tested the umbrella repository approach to make sure the setup with multiple nested *directories* works. (It does)

This was the layout:
  * px4ros :umbrella: 
    * Firmware
    * euroc_simulator
      * rotors_control
      * rotors_...
      * ...
    * mav_comm
    * catkin_simple
    * glog_catkin
  * mavros
  * ...

**Comment by thomasgubler** [comment]
@AndreasAntener this PR is waiting for your ok to merge

File: launch/mavros_sitl.launch [hunk 1]
```diff
@@ -1,7 +1,9 @@
 <launch>
-	<!-- vim: set ft=xml noet : -->
-	<!-- example launch script for PX4 based FCU's -->
+<!-- vim: set ft=xml noet : -->
+<!-- example launch script for PX4 based FCU's -->
 
+<arg name="ns"/>
```
Comment (AndreasAntener) [inline launch/mavros_sitl.launch]: Would it make sense to have a default here? e.g. "/"

**Comment by AndreasAntener** [comment]
:+1: works for me! I was only wondering about maybe adding default namespace for the mavros include.

If you do an umbrella project, "containers" should also be in there (for CI)

File: launch/mavros_sitl.launch (hunk 1 above)
Comment (thomasgubler) [inline launch/mavros_sitl.launch]: maybe for people who run mavros standalone for testing? added in b311f7302b508d1ed7fd3cfecd0bfbe47b324bde

**Comment by thomasgubler** [comment]
thanks for the test!
rebased, merging

**Pull request merged**